src/daemon/default_vm_image_vault.cpp

```diff
@@ -344,50 +344,44 @@ mp::VMImage mp::DefaultVMImageVault::fetch_image(const FetchType& fetch_type, co
                 {
                     return finalize_image_records(query, record.image, id);
                 }
-                else
-                {
-                    *source_image = record.image;
-                }
+            }
+
+            auto running_future = get_image_future(id);
```

Comment:
This isn't really new except for I moved the whole chunk of the code out of an `else` block (and removed the `else` block) since we need to run through the following code regardless.  The diff is really bad about showing what I did here...

Reply:
In this and the other questions, I understood that the code block was only moved, but wanted to confirm the implications of executing the composite operation in the different conditions are as expected.

Reply:
Ok, to answer your original inquiry, yes, it checks to see if there is already an in-progress download of the same URL.  And yes, this happens if there is no already existing prepared image or if the lastUpdated header for the URL is newer than what has been saved, implicating that a new download is needed.

Reply:
OK thanks